bigquery_etl/parse_udf.py

```diff
@@ -41,12 +41,31 @@ def from_file(filepath):
         with open(filepath) as f:
             text = f.read()
 
+        name = basename.replace(".sql", "")
+        dataset = os.path.basename(dirpath)
+
+        try:
+            udf = RawUdf.from_text(text, dataset, name)
+        except ValueError as e:
+            raise ValueError(str(e) + f" in {filepath}")
+
+        udf.filepath = filepath
+        return udf
+
+    @staticmethod
+    def from_text(text, dataset, name, is_defined=True):
```

Comment:
this should specify `filepath` as an argument, rather than setting it afterward:
```suggestion
        try:
            return RawUdf.from_text(text, dataset, name, filepath)
        except ValueError as e:
            raise ValueError(str(e) + f" in {filepath}")

    @staticmethod
    def from_text(text, dataset, name, filepath=None, is_defined=True):
```